src/win/pty.cc

```diff
@@ -131,6 +131,23 @@ void throw_winpty_error(const char *generalMsg, winpty_error_ptr_t error_ptr) {
   winpty_error_free(error_ptr);
 }
 
+static NAN_METHOD(PtyGetExitCode) {
+  Nan::HandleScope scope;
+
+  if (info.Length() != 1
+    || !info[0]->IsNumber()) // pidHandle
+  {
+    Nan::ThrowError("Usage: pty.getExitCode(pidHandle)");
+    return;
+  }
+
+  uint32_t pidHandle = info[0]->Uint32Value();
+  DWORD exitCode = 0;
+  GetExitCodeProcess((HANDLE)pidHandle, &exitCode);
```

Comment:
Have you tested this with Windows64 and node 64 bit? Can't test it myself right now but I think the pointer extension always with zeros is wrong here. https://msdn.microsoft.com/en-us/library/windows/desktop/aa384203%28v=vs.85%29.aspx states that a HANDLE needs to be sign-extended for 32 to 64 bit conversion. Imho `pidHandle` should be signed not unsigned.